src/Domain/DomainHelpers.cpp

```diff
@@ -261,6 +261,66 @@ obtain_correspondence_between_blocks(
                                               nhbr_align);
 }
 
+template <size_t VolumeDim>
+std::vector<std::array<size_t, two_to_the(VolumeDim)>> corners_from_two_maps(
+    const std::unique_ptr<domain::CoordinateMapBase<
+        Frame::Logical, Frame::Inertial, VolumeDim>>& map1,
+    const std::unique_ptr<
+        domain::CoordinateMapBase<Frame::Logical, Frame::Inertial, VolumeDim>>&
+        map2) noexcept {
+  std::array<size_t, two_to_the(VolumeDim)> corners_for_block1;
+  std::array<size_t, two_to_the(VolumeDim)> corners_for_block2;
+  for (VolumeCornerIterator<VolumeDim> vci{}; vci; ++vci) {
+    gsl::at(corners_for_block1, vci.local_corner_number()) =
+        vci.local_corner_number();
+  }
+  // Fill corners_for_block2 based off which corners are shared with map1.
+  for (VolumeCornerIterator<VolumeDim> vci_map2{}; vci_map2; ++vci_map2) {
+    const auto mapped_coords2 =
+        (*map2)(tnsr::I<double, VolumeDim, Frame::Logical>{
+            vci_map2.coords_of_corner()});
+    double num_unshared_corners = 0;
+    // Check each corner of map1 to see if it shares any corners with map2.
+    for (VolumeCornerIterator<VolumeDim> vci_map1{}; vci_map1; ++vci_map1) {
+      const auto mapped_coords1 =
+          (*map1)(tnsr::I<double, VolumeDim, Frame::Logical>{
+              vci_map1.coords_of_corner()});
+      double max_separation = 0.0;
+      for (size_t j = 0; j < VolumeDim; j++) {
+        max_separation =
+            std::max(max_separation,
+                     std::abs(mapped_coords2.get(j) - mapped_coords1.get(j)));
+      }
+      // If true, then the mapped_coords lie on top of one another.
+      // This corner of map2 is assigned the same number as that of map1.
+      if (max_separation < 1.0e-6) {
+        gsl::at(corners_for_block2, vci_map2.local_corner_number()) =
+            vci_map1.local_corner_number();
+        break;
+      }
+      // Otherwise, a new number is assigned to this corner.
+      else {
+        gsl::at(corners_for_block2, vci_map2.local_corner_number()) =
+            two_to_the(VolumeDim) + num_unshared_corners;
+        num_unshared_corners++;
```

Comment:
What's not clear to me is how if there are many blocks you don't get duplicate corner numbers since `num_unshared_corners` always starts at zero. Does that not matter because you just treat each pair of blocks independent of all the others? I.e. you don't create a global corner numbering? If so, I think having a comment along the lines of:
```
// Set num_unshared_corners to zero because we only need a local corner numbering scheme. 
// This is because our algorithm treats each pair of blocks independently to create the 
// orientation (if any) between them.
```
where `num_unshared_corners` is created would be useful

Reply:
That is correct :smile: there is never a global numbering created, since relative orientations only care about two blocks at a time.

Reply:
Sweet :D Thank you!